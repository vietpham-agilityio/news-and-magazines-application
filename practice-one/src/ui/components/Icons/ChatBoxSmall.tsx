interface IProps {
    color?: string;
  }

  export const ChatBoxSmall = ({ color="#FFF"}: IProps) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width={15}
    height={13}
    fill="none"
    data-testid="chat-box-icon"
  >
    <path
      fill={color}
      d="M4.938 5.438c.464 0 .847.41.847.875a.858.858 0 0 1-.848.875.864.864 0 0 1-.875-.875c0-.465.383-.875.875-.875Zm3.062 0c.465 0 .848.41.848.875 0 .437-.41.847-.848.847-.465 0-.875-.383-.875-.848 0-.492.383-.875.875-.875Zm3.063 0c.464 0 .874.41.847.875a.858.858 0 0 1-.848.875.881.881 0 0 1-.874-.875c0-.465.382-.875.874-.875ZM8 .625c3.855 0 6.973 2.598 6.973 5.715S11.855 12 8 12a8.402 8.402 0 0 1-2.543-.383c-.82.574-2.133 1.258-3.8 1.258-.274 0-.52-.137-.602-.41-.082-.246-.055-.52.136-.711 0 0 .848-.93 1.258-1.996C1.547 8.8 1 7.598 1 6.313 1 3.167 4.117.625 8 .625Zm0 10.063c3.117 0 5.633-1.942 5.633-4.376 0-2.406-2.543-4.375-5.66-4.375-3.118 0-5.66 1.97-5.66 4.375 0 1.176.574 2.051 1.066 2.57l.574.602-.3.766a5.116 5.116 0 0 1-.548 1.094 6.104 6.104 0 0 0 1.586-.793l.52-.383.629.191a7.132 7.132 0 0 0 2.16.329Z"
    />
  </svg>
);
