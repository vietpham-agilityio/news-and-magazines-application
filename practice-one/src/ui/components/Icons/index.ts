export * from '@/ui/components/Icons/User';
export * from '@/ui/components/Icons/Plus';
export * from '@/ui/components/Icons/Home';
export * from '@/ui/components/Icons/File';
export * from '@/ui/components/Icons/Close';
export * from '@/ui/components/Icons/Circle';
export * from '@/ui/components/Icons/Search';
export * from '@/ui/components/Icons/SaveTag';
export * from '@/ui/components/Icons/MailBox';
export * from '@/ui/components/Icons/Twitter';
export * from '@/ui/components/Icons/ChatBox';
export * from '@/ui/components/Icons/Ellipse';
export * from '@/ui/components/Icons/Schedule';
export * from '@/ui/components/Icons/ArrowTop';
export * from '@/ui/components/Icons/ThreeDots';
export * from '@/ui/components/Icons/Triangle';
export * from '@/ui/components/Icons/Instagram';
export * from '@/ui/components/Icons/BurgerMenu';
export * from '@/ui/components/Icons/ChevronsUp';
export * from '@/ui/components/Icons/ArrowLeft';
export * from '@/ui/components/Icons/ArrowBack';
export * from '@/ui/components/Icons/ArrowRight';
export * from '@/ui/components/Icons/PaperPlane';
export * from '@/ui/components/Icons/ArrowBottom';
export * from '@/ui/components/Icons/ChatBoxSmall';
export * from '@/ui/components/Icons/SaveTagSmall';