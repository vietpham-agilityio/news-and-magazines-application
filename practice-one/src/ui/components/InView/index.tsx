'use client';

import { InView } from 'react-intersection-observer';

interface IProps {
  children: React.ReactNode;
}

const InViewWrapper = ({ children }: IProps) => {
  return (
    <InView triggerOnce threshold={1}>
      {({ ref, inView }) => (
        <div ref={ref} className="w-full">
          {inView && children}
        </div>
      )}
    </InView>
  );
};

export default InViewWrapper;
