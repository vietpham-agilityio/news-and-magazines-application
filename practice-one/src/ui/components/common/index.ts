export * from '@/ui/components/common/Tag';
export * from '@/ui/components/common/Menu';
export * from '@/ui/components/common/Title';
export * from '@/ui/components/common/Input';
export * from '@/ui/components/common/Avatar';
export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/BoxIcon';
export * from '@/ui/components/common/SubMenu';
export * from '@/ui/components/common/TextArea';
export * from '@/ui/components/common/Typography';
export * from '@/ui/components/common/Information';
