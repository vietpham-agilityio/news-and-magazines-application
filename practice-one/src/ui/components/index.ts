// common
export * from '@/ui/components/common';

// shared
export * from '@/ui/components/Brand';
export * from '@/ui/components/NavItem';
export * from '@/ui/components/Comment';
export * from '@/ui/components/Carousel';
export * from '@/ui/components/SearchBar';
export * from '@/ui/components/PaperPost';
export * from '@/ui/components/BurgerMenu';
export * from '@/ui/components/Pagination';
export * from '@/ui/components/CommentForm';
export * from '@/ui/components/BreadCrumbs';
export * from '@/ui/components/TitleSection';
export * from '@/ui/components/SubmitButton';
export * from '@/ui/components/Card/CardSingle';
export * from '@/ui/components/Skeleton/Comment';
export * from '@/ui/components/Skeleton/CardPost';
export * from '@/ui/components/Skeleton/BadgeGroup';
export * from '@/ui/components/Skeleton/DetailPost';
export * from '@/ui/components/Skeleton/CardVertical';
export * from '@/ui/components/Skeleton/CardCategory';
export { default as CardPost } from '@/ui/components/Card/CardPost';
export { default as CardCategory } from '@/ui/components/Card/CardCategory';