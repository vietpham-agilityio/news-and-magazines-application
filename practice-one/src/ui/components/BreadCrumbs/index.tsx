// components
import Link from 'next/link';
import { Typography } from '@/ui/components';

// types
import { FontWeight, IBreadCrumbItem, Size } from '@/types';

// icons
import { ArrowRight } from '@/ui/components/Icons';

interface IProps {
  listBreadCrumb: IBreadCrumbItem[];
}

export const BreadCrumbs = ({ listBreadCrumb }: IProps) => (
  <div className="flex items-center">
    {listBreadCrumb.map((item, index) => {
      const { path, name } = item;

      return (
        <div key={name} className="flex items-center">
          <Link href={path}>
            <Typography
              tag="h2"
              textSize={Size.XS}
              weight={FontWeight.Normal}
              additionalClasses={`
              ${
                index === listBreadCrumb.length - 1
                  ? 'w-57.5 text-dark-90 h-5 truncate text-ellipsis overflow-hidden sm:w-full'
                  : ' text-dark-100 hover:text-primary-100'
              }`}
            >
              {name}
            </Typography>
          </Link>
          {index !== listBreadCrumb.length - 1 && (
            <i className="mx-2">
              <ArrowRight />
            </i>
          )}
        </div>
      );
    })}
  </div>
);
