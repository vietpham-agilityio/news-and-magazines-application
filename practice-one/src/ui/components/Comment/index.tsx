// static image
import ImageStore from '@/constants/images';

// components
import { Avatar, Button, Typography } from '@/ui/components';

// icons
import { ArrowBack, Schedule } from '@/ui/components/Icons';

// types
import { FontWeight, Size } from '@/types';

interface IProps {
  userName: string;
  content: string;
  publicDate?: string;
}

export const Comment = ({ userName, content, publicDate }: IProps) => (
  <div className="bg-white-90 p-4 rounded-xl">
    { publicDate ? (
      <div className="flex flex-1 px-1 mb-4">
        <Avatar
          width={60}
          height={60}
          url={ImageStore.AvatarUser}
          alt="Author this comment"
        />
        <div className="flex flex-1 items-center justify-between">
          <div className="flex flex-1 justify-between">
            <div className="flex flex-col h-full ml-2.5">
              <Typography
                tag="h2"
                textSize={Size.SM}
                additionalClasses="text-dark-100"
              >
                {userName}
              </Typography>
              <div className="flex gap-2">
                <Schedule />
                <Typography
                  tag="h3"
                  weight={FontWeight.Medium}
                  additionalClasses="text-dark-75 text-p leading-5"
                >
                  {publicDate}
                </Typography>
              </div>
            </div>
          </div>
        </div>
      </div>
    ) : (
      <Typography tag="h2" textSize={Size.SM} additionalClasses="text-dark-100">
        {userName}
      </Typography>
    )}
    <Typography
      tag="h3"
      weight={FontWeight.Light}
      additionalClasses={`${publicDate ? 'text-xsm text-black-75 leading-5' : 'text-p text-dark-75'}`}
    >
      {content}
    </Typography>
  </div>
);
