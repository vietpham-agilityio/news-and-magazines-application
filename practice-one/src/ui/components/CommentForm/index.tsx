// components
import { createComment } from '@/services';
import { Input, SubmitButton, TextArea } from '@/ui/components';

// icons
import { ChatBoxSmall } from '@/ui/components/Icons';

interface IProps {
  postId: string;
}

export const CommentForm = ({postId}: IProps) => {

return (
  <form action={createComment}>
    <div className="grid grid-cols-12 grid-rows-3 lg:gap-6">
      <div className="row-span-3 col-span-12 lg:col-span-6">
        <input name="postId" type="hidden" value={postId} />
        <div className="mb-5 row-span-1">
          <Input label="Name" name="name" type="text" />
        </div>
        <div className="mb-5 row-span-1">
          <Input label="Email" name="email" type="email" />
        </div>
        <div className="mb-5 row-span-1">
          <Input label="Website" name="website" type="text" />
        </div>
      </div>
      <div className="row-span-3 col-span-12 lg:col-span-6">
        <TextArea name='content' label="Comment" placeholder="What do you think?" />
      </div>
    </div>
    <div className="flex w-full justify-end mt-7 mb-30 sm:mt-10 sm:mb-12 lg:mt-0">
      <SubmitButton
        label="Send Comment"
        leftIcon={<ChatBoxSmall />}
      />
    </div>
  </form>
)}
