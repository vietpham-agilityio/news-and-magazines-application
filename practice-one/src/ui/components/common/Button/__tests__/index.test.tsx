import { fireEvent, render, screen } from '@testing-library/react';

// component
import { Button } from '@/ui/components';
import { ArrowLeft, ArrowRight } from '@/ui/components/Icons';

let renderButton: any;
const onClickMock = jest.fn();

describe('Button component', () => {
  beforeEach(() => {
    renderButton = render(
      <Button
        isMarginBottom
        label="Subcriber"
        textColor="text-dark-100"
        bgColor="bg-primary-100"
        onClick={onClickMock}
      />
    );
  });

  afterEach(() => {
    renderButton.unmount();
  });

  it('Button should render macth snapshot', () => {
    expect(renderButton.asFragment()).toMatchSnapshot();
  });

  it('Button should render label alt value to macth value passed', () => {
    const labelValue = screen.getByText('Subcriber');

    expect(labelValue).toBeInTheDocument();
  });

  it('Button should trigger event when clicked', () => {
    const buttonElement = screen.getByRole('button');

    fireEvent.click(buttonElement);
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('Button should contain left icon', () => {
    const { getByTestId } = render(
      <Button
        isMarginBottom
        label="Subcriber"
        textColor="text-dark-100"
        bgColor="bg-primary-100"
        leftIcon={<ArrowLeft />}
        onClick={onClickMock}
      />
    );

    const iconElement = getByTestId('arrow-left');

    expect(iconElement).toBeInTheDocument();
  });

  it('Button should contain right icon', () => {
    const { getByTestId } = render(
      <Button
        isMarginBottom
        label="Subcriber"
        textColor="text-dark-100"
        bgColor="bg-primary-100"
        rightIcon={<ArrowRight />}
        onClick={onClickMock}
      />
    );

    const iconElement = getByTestId('arrow-right');

    expect(iconElement).toBeInTheDocument();
  });

  it('Button should render disabled state macth snapshot', () => {
    const { asFragment } = render(
      <Button
        isDisabled
        isMarginBottom
        label="Subcriber"
        textColor="text-dark-100"
        bgColor="bg-primary-100"
        rightIcon={<ArrowRight />}
        onClick={onClickMock}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
