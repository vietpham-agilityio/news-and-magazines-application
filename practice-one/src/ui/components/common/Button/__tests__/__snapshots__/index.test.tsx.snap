// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button should render disabled state macth snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="button-element"
    class="flex items-center rounded-xl bg-primary-100 py-2.5 undefined pr-4 pl-6 hover:cursor-not-allowed hover:opacity-70"
  >
    <h2
      aria-level="2"
      class="text-xsm font-medium text-dark-100 leading-5"
      role="heading"
    >
      Subcriber
    </h2>
    <i
      class="ml-2"
    >
      <svg
        data-testid="arrow-right"
        fill="none"
        height="11"
        width="7"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.25 5.5a.866.866 0 0 1-.246.629l-4.375 4.375a.843.843 0 0 1-1.23 0 .843.843 0 0 1 0-1.23L4.144 5.5.398 1.754a.843.843 0 0 1 0-1.23.843.843 0 0 1 1.23 0l4.376 4.374a.848.848 0 0 1 .246.602Z"
          fill="#3E3232"
          fill-opacity="0.5"
        />
      </svg>
    </i>
  </button>
</DocumentFragment>
`;

exports[`Button component Button should render macth snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="button-element"
    class="flex items-center rounded-xl bg-primary-100 py-2.5 undefined undefined hover:opacity-80"
  >
    <h2
      aria-level="2"
      class="text-xsm font-medium text-dark-100 leading-5"
      role="heading"
    >
      Subcriber
    </h2>
  </button>
</DocumentFragment>
`;
